### ISSUE TITLE
BUG: TestMergeDtypes.test_merge_on_ints_floats_warning fails (Linux/openSUSE/Factory)

### ISSUE THREAD

**Issue by mcepl** [pandas-dev/pandas#34299]
- [X] I have checked that this issue has not already been reported (cannot find any ``test_merge`` issue)

- [X] I have confirmed this bug exists on the latest version of pandas (testing on 1.0.3, which is the latest released).

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas. (This happened while building the package for openSUSE, so we care to use only officially released versions).

#### Problem description

While running the test suite, this one test fails consistently:
```
[  363s] + export PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-pandas-1.0.3-0.i386/usr/lib/python3.8/site-packages
[  363s] + PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-pandas-1.0.3-0.i386/usr/lib/python3.8/site-packages
[  363s] + xvfb-run py.test-3.8 -n auto -v /home/abuild/rpmbuild/BUILDROOT/python-pandas-1.0.3-0.i386/usr/lib/python3.8/site-packages/pandas/tests -k 'not test_oo_optimizable and not test_encode_non_c_locale and not test_maybe_promote_int_with_int'
[  368s] ============================= test session starts ==============================
[  368s] platform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
[  368s] cachedir: .pytest_cache
[  368s] hypothesis profile 'ci' -> deadline=timedelta(milliseconds=500), suppress_health_check=[HealthCheck.too_slow], database=DirectoryBasedExampleDatabase('/home/abuild/rpmbuild/BUILD/pandas-1.0.3/.hypothesis/examples')
[  368s] rootdir: /home/abuild/rpmbuild
[  368s] plugins: forked-1.1.3, hypothesis-5.11.0, xdist-1.32.0, mock-3.1.0
[ 1070s] =================================== FAILURES ===================================
[ 1070s] ______________ TestMergeDtypes.test_merge_on_ints_floats_warning _______________
[ 1070s] [gw0] linux -- Python 3.8.2 /usr/bin/python3
[ 1070s] 
[ 1070s] self = <pandas.tests.reshape.merge.test_merge.TestMergeDtypes object at 0x9419e988>
[ 1070s] 
[ 1070s]     def test_merge_on_ints_floats_warning(self):
[ 1070s]         # GH 16572
[ 1070s]         # merge will produce a warning when merging on int and
[ 1070s]         # float columns where the float values are not exactly
[ 1070s]         # equal to their int representation
[ 1070s]         A = DataFrame({"X": [1, 2, 3]})
[ 1070s]         B = DataFrame({"Y": [1.1, 2.5, 3.0]})
[ 1070s]         expected = DataFrame({"X": [3], "Y": [3.0]})
[ 1070s]     
[ 1070s]         with tm.assert_produces_warning(UserWarning):
[ 1070s]             result = A.merge(B, left_on="X", right_on="Y")
[ 1070s]             tm.assert_frame_equal(result, expected)
[ 1070s]     
[ 1070s]         with tm.assert_produces_warning(UserWarning):
[ 1070s]             result = B.merge(A, left_on="Y", right_on="X")
[ 1070s]             tm.assert_frame_equal(result, expected[["Y", "X"]])
[ 1070s]     
[ 1070s]         # test no warning if float has NaNs
[ 1070s]         B = DataFrame({"Y": [np.nan, np.nan, 3.0]})
[ 1070s]     
[ 1070s]         with tm.assert_produces_warning(None):
[ 1070s]             result = B.merge(A, left_on="Y", right_on="X")
[ 1070s] >           tm.assert_frame_equal(result, expected[["Y", "X"]])
[ 1070s] 
[ 1070s] ../../BUILDROOT/python-pandas-1.0.3-0.i386/usr/lib/python3.8/site-packages/pandas/tests/reshape/merge/test_merge.py:1484: 
[ 1070s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 1070s] 
[ 1070s] self = <contextlib._GeneratorContextManager object at 0x9419e340>, type = None
[ 1070s] value = None, traceback = None
[ 1070s] 
[ 1070s]     def __exit__(self, type, value, traceback):
[ 1070s]         if type is None:
[ 1070s]             try:
[ 1070s] >               next(self.gen)
[ 1070s] E               AssertionError: Caused unexpected warning(s): [('RuntimeWarning', RuntimeWarning('invalid value encountered in equal'), '/home/abuild/rpmbuild/BUILDROOT/python-pandas-1.0.3-0.i386/usr/lib/python3.8/site-packages/pandas/core/reshape/merge.py', 1091)]
[ 1070s] 
[ 1070s] /usr/lib/python3.8/contextlib.py:120: AssertionError
```

#### Expected Output

All tests pass.

#### Output of ``pd.show_versions()``

<details>

[  403s] + python3 -c 'import pandas; print(pandas.show_versions())'
[  404s] /usr/lib/python3.8/site-packages/setuptools/version.py:1: UserWarning: Module pandah
[  404s]   import pkg_resources
[  404s] 
[  404s] INSTALLED VERSIONS
[  404s] ------------------
[  404s] commit           : None
[  404s] python           : 3.8.2.final.0
[  404s] python-bits      : 32
[  404s] OS               : Linux
[  404s] OS-release       : 5.6.12-1-default
[  404s] machine          : i686
[  404s] processor        : i686
[  404s] byteorder        : little
[  404s] LC_ALL           : en_US.UTF-8
[  404s] LANG             : en_US.UTF-8
[  404s] LOCALE           : en_US.UTF-8
[  404s] 
[  404s] pandas           : 1.0.3
[  404s] numpy            : 1.18.4
[  404s] pytz             : 2020.1
[  404s] dateutil         : 2.8.1
[  404s] pip              : None
[  404s] setuptools       : 44.0.0
[  404s] Cython           : 0.29.17
[  404s] pytest           : 5.3.5
[  404s] hypothesis       : 5.11.0
[  404s] sphinx           : None
[  404s] blosc            : None
[  404s] feather          : None
[  404s] xlsxwriter       : 1.2.7
[  404s] lxml.etree       : 4.5.0
[  404s] html5lib         : None
[  404s] pymysql          : None
[  404s] psycopg2         : None
[  404s] jinja2           : 2.11.2
[  404s] IPython          : None
[  404s] pandas_datareader: None
[  404s] bs4              : 4.9.0
[  404s] bottleneck       : None
[  404s] fastparquet      : None
[  404s] gcsfs            : None
[  404s] lxml.etree       : 4.5.0
[  404s] matplotlib       : None
[  404s] numexpr          : None
[  404s] odfpy            : None
[  404s] openpyxl         : 3.0.3
[  404s] pandas_gbq       : None
[  404s] pyarrow          : None
[  404s] pytables         : None
[  404s] pytest           : 5.3.5
[  404s] pyxlsb           : None
[  404s] s3fs             : None
[  404s] scipy            : None
[  404s] sqlalchemy       : 1.3.16
[  404s] tables           : None
[  404s] tabulate         : None
[  404s] xarray           : None
[  404s] xlrd             : 1.2.0
[  404s] xlwt             : 1.3.0
[  404s] xlsxwriter       : 1.2.7
[  404s] numba            : None
[  404s] None

</details>

[Complete build log with all details of the process](https://github.com/pandas-dev/pandas/files/4664063/_log.txt.zip)

**Comment by TomAugspurger** [comment]
@mcepl can you try again? Sometimes this type of failure indicates a warning leaking from another test.

Given that this seems to be somewhat platform specific, are you able to investigate it?

**Comment by mcepl** [comment]
> @mcepl can you try again? Sometimes this type of failure indicates a warning leaking from another test.

Yes, it seems that THIS failed test got somehow fixed in 1.1.1.
